go/worker/common/p2p/dispatch.go

```diff
@@ -153,13 +156,15 @@ func (h *topicHandler) dispatchMessage(peerID core.PeerID, m *queuedMsg, isIniti
 	// Perhaps this should reject the message, but it is possible that
 	// the local node is just behind.  This does result in stale messages
 	// getting retried though.
-	if err = h.handler.AuthenticatePeer(m.from, m.msg); err != nil {
-		return err
-	}
+	for _, handler := range h.handlers {
```

Comment:
No lock on the handlers? The `topicHandler` should probably have a RW lock and the RLock should be taken here while a Lock should be taken in `RegisterHandler`.